src/envoy/mixer/quota_config.cc

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/mixer/quota_config.h"
+
+using ::istio::mixer_client::Attributes;
+using ::istio::mixer::v1::config::client::AttributeMatch;
+using ::istio::mixer::v1::config::client::QuotaSpec;
+
+namespace Envoy {
+namespace Http {
+namespace Mixer {
+namespace {
+
+bool MatchAttributes(const AttributeMatch& match,
+                     const Attributes& attributes) {
+  for (const auto& map_it : match.clause()) {
+    // map is attribute_name to StringMatch.
+    const std::string& name = map_it.first;
+    const auto& match = map_it.second;
+
+    // Check if required attribure exists with string type.
+    const auto& it = attributes.attributes.find(name);
+    if (it == attributes.attributes.end() ||
+        it->second.type != Attributes::Value::STRING) {
+      return false;
+    }
+    const std::string& value = it->second.str_v;
+
+    switch (match.match_type_case()) {
+      case ::istio::proxy::v1::config::StringMatch::kExact:
+        if (value != match.exact()) {
+          return false;
+        }
+        break;
+      case ::istio::proxy::v1::config::StringMatch::kPrefix:
+        if (value.length() < match.prefix().length() ||
+            value.compare(0, match.prefix().length(), match.prefix()) != 0) {
+          return false;
+        }
+        break;
+      case ::istio::proxy::v1::config::StringMatch::kRegex:
+        // TODO: support regex
+        return false;
+        break;
+      default:
+        break;
```

Comment:
Let's not allow this fall through and return `true`?

Reply:
Should an `unknown` match type result in a valid match? Seems like the default behavior should be to ignore that case.

Reply:
@ayj I agreed with.  We should ignore "Unknown" type

Reply:
No, there shouldn't be a unknown clause (it should be removed) and it shouldn't go through. You treated "unimplemented" regex as "no match" above, so the `unknown` should be same. Also leave a comment here.

Reply:
OK, I changed "unimplemented" as ignored.  so regex and not_set case are all ignored.

Reply:
+ 1 for comment explanation rationale. 

Really, this type of error should be caught earlier by validation before proxy receives the config. 

Shouldn't adding unoptimized regex support be fairly straightforward? e.g.

```c++
if (!std::regex_match(value, match.regex())) {
    return false
}
```